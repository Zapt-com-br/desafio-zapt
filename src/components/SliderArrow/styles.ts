import styled, { css } from 'styled-components';
import { lighten } from 'polished';
import media from 'styled-media-query';

import { SliderArrowProps } from '.';

const wrapperModifiers = {
  left: () => css`
    transform: rotate(90deg);
    left: -0.4rem;

    svg {
      margin-top: 0.8rem;
    }

    ${media.greaterThan('large')`
      left: -0.9rem;

      svg {
        margin-top: 1.5rem;
      }
    `}
  `,

  right: () => css`
    transform: rotate(-90deg);
    right: -0.4rem;

    svg {
      margin-top: 0.8rem;
    }

    ${media.greaterThan('large')`
      right: -0.9rem;

      svg {
        margin-top: 1.5rem;
      }
    `}
  `,
};

export const Wrapper = styled.button<SliderArrowProps>`
  ${({ theme, direction }) => css`
    background-color: ${theme.colors.white};
    color: ${theme.colors.primaryDarken};
    width: 6.3rem;
    height: 3rem;
    border-top-left-radius: 6.4rem;
    border-top-right-radius: 6.4rem;
    cursor: pointer;
    border: 0;
    outline: none;
    position: absolute;
    top: 35%;
    bottom: 0;
    transition: background-color ${theme.transition.default},
      color ${theme.transition.fast}, visibility ${theme.transition.default};
    z-index: ${theme.layers.base};

    &:hover {
      background-color: ${theme.colors.primary};
      color: ${theme.colors.white};
    }

    &:focus {
      box-shadow: 0 0 0 3px ${lighten(0.2, theme.colors.primary)};
    }

    &:active {
      background: ${lighten(0.2, theme.colors.primary)};
    }

    svg {
      width: 2rem;
      height: 2rem;
      transform: rotate(90deg);
    }

    ${media.greaterThan('large')`
      width: 8.4rem;
      height: 4.2rem;
      border-top-left-radius: 8.5rem;
      border-top-right-radius: 8.5rem;

      svg {
        width: 2.4rem;
        height: 2.4rem;
      }
    `}

    ${!!direction && wrapperModifiers[direction!]()};
  `}
`;
