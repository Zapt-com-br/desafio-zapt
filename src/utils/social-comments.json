{
  "title": "O que est√£o falando nas redes sociais",
  "comments": [
    {
      "id": 1,
      "comment": "N√£o da para parar de comer.. esse final de semana pedi 20 pasteis da Ilma e acabaram TODOS em um s√≥ dia... O pessoal aqui de casa ficou viciado de um n√≠vel que terei que comprar toda semana... Assim n√£o tem academia que secura rsrsrsrs Obrigado Zapt.",
      "name": "ferri-gui",
      "image": "/images/social-comments/ferri-gui.svg"
    },
    {
      "id": 2,
      "comment": "Estava precisando achar algum lugar que vendia produtos que v√™em direto da fazenda... a Zapt salvou.. e ainda paquei um pre√ßo muito mais barato que supermecado üôå",
      "name": "paolaC",
      "image": "/images/social-comments/paola.svg"
    },
    {
      "id": 3,
      "comment": "Todos os vinhos da vin√≠cula Bueno est√£o dispon√≠veis no catalogo da Zapt a um pre√ßo incriv√©l... n√£o percam essa oportunidade...",
      "name": "Galv√£o",
      "image": "/images/social-comments/galvao.svg"
    },
    {
      "id": 4,
      "comment": "Nunca tinha achado os produtos na Lush depois que ela saiu do Brasil.. s√≥ a Zapt para conseguir importalos a pre√ßo mais barato do que vendiam aqui! Muito obrigado, recomendo sempre a todos os amigos.",
      "name": "LiciaR",
      "image": "/images/social-comments/liciar.svg"
    },
    {
      "id": 5,
      "comment": "Ontem chegou uma pizza que comprei na Zapt e me lembrou as que minha av√≥ fazia... sensacional!",
      "name": "shitz",
      "image": "/images/social-comments/shitz.svg"
    },
    {
      "id": 6,
      "comment": "Nossa cada dia tem uma oferta mais legal do que a outra.. Al√©m de produtos de qualidade o pr√©co √© imbat√≠vel",
      "name": "rafael",
      "image": "/images/social-comments/rafael.svg"
    },
    {
      "id": 7,
      "comment": "Achei uma promo√ß√£o incr√≠vel de roupinha de beb√™, foi o que salvou o enxoval do Pedro.",
      "name": "gust",
      "image": "/images/social-comments/gust.svg"
    },
    {
      "id": 8,
      "comment": "Meu dono finalmente conseguiu achar uma comida que eu gosto! Agora vou conseguir engordar e ser feliz.",
      "name": "Cookie",
      "image": "/images/social-comments/cookie.svg"
    },
    {
      "id": 9,
      "comment": "N√£o conhecia a Zapt.. um amigo meu indicou... falou para eu dar uma olhada nos pre√ßos das cervejas artesanais.. incr√≠vel.",
      "name": "duque",
      "image": "/images/social-comments/duque.svg"
    }
  ]
}
