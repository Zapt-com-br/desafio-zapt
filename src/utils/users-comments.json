{
  "title": "O que as pessoas estão achando",
  "comments": [
    {
      "id": 1,
      "image": "/images/user-comments/image-1.png",
      "comment": "Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal",
      "user": {
        "avatar": "/images/user-comments/patricia.svg",
        "name": "Patricia Godoy",
        "city": "São Paulo",
        "state": "São Paulo"
      }
    },
    {
      "id": 2,
      "image": "/images/user-comments/image-2.png",
      "comment": "Depois que você começa a usar produtos de qualidade é bem difícil voltar atrás...",
      "user": {
        "avatar": "/images/user-comments/marcinho.svg",
        "name": "Marcinho Ribeiro",
        "city": "Belo Horizonte",
        "state": "Minas Gerais"
      }
    },
    {
      "id": 3,
      "image": "/images/user-comments/image-1.png",
      "comment": "Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal",
      "user": {
        "avatar": "/images/user-comments/patricia.svg",
        "name": "Patricia Godoy",
        "city": "São Paulo",
        "state": "São Paulo"
      }
    },
    {
      "id": 4,
      "image": "/images/user-comments/image-2.png",
      "comment": "Depois que você começa a usar produtos de qualidade é bem difícil voltar atrás...",
      "user": {
        "avatar": "/images/user-comments/marcinho.svg",
        "name": "Marcinho Ribeiro",
        "city": "Belo Horizonte",
        "state": "Minas Gerais"
      }
    }
  ]
}
