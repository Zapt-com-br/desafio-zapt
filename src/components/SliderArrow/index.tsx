import { ButtonHTMLAttributes } from 'react';

import * as S from './styles';

export type SliderArrowProps = {
  direction: 'right' | 'left';
  label?: string;
} & ButtonHTMLAttributes<HTMLButtonElement>;

const SliderArrow = ({ direction, label, ...props }: SliderArrowProps) => (
  <S.Wrapper {...props} direction={direction}>
    <svg fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
      <path
        d="M8.625 5.25l6.75 6.75-6.75 6.75"
        stroke="currentColor"
        strokeWidth="2"
        strokeLinecap="round"
        strokeLinejoin="round"
        role="img"
        aria-label={label || `arrow ${direction}`}
      />
    </svg>
  </S.Wrapper>
);

export default SliderArrow;
