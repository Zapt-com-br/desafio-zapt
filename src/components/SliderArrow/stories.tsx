import { Story, Meta } from '@storybook/react/types-6-0';

import SliderArrow from '.';

export default {
  title: 'SliderArrow',
  component: SliderArrow,
  parameters: {
    backgrounds: {
      default: 'dark',
    },
  },
} as Meta;

export const Default: Story = () => (
  <div
    style={{
      width: '100%',
      marginTop: '2.4rem',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      position: 'relative',
    }}
  >
    <SliderArrow direction="left" />
    <SliderArrow direction="right" />
  </div>
);
