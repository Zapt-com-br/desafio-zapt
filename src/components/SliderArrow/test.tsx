import { render, screen } from 'utils/test-utils';
import userEvent from '@testing-library/user-event';

import SliderArrow from '.';

describe('<SliderArrow />', () => {
  it('should render right arrow', () => {
    const { container } = render(<SliderArrow direction="right" />);

    expect(
      screen.getByRole('img', { name: /arrow right/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render left arrow', () => {
    const { container } = render(<SliderArrow direction="left" />);

    expect(
      screen.getByRole('img', { name: /arrow left/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with custom label arrow', () => {
    render(<SliderArrow direction="right" label="my custom label" />);

    expect(
      screen.getByRole('img', { name: /my custom label/i })
    ).toBeInTheDocument();
  });

  it('should dispatch onClick when clicked', () => {
    const onClick = jest.fn();

    render(<SliderArrow direction="right" onClick={onClick} />);

    userEvent.click(screen.getByRole('img', { name: /arrow right/i }));

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
