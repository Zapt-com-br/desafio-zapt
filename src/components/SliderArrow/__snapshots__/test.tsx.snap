// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SliderArrow /> should render left arrow 1`] = `
.c0 {
  background-color: #ffffff;
  color: #011fbb;
  width: 6.3rem;
  height: 3rem;
  border-top-left-radius: 6.4rem;
  border-top-right-radius: 6.4rem;
  cursor: pointer;
  border: 0;
  outline: none;
  position: absolute;
  top: 40%;
  bottom: 0;
  -webkit-transition: background-color 0.3s ease-in-out,color 0.1s ease-in-out,visibility 0.3s ease-in-out;
  transition: background-color 0.3s ease-in-out,color 0.1s ease-in-out,visibility 0.3s ease-in-out;
  z-index: 10;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  left: -0.4rem;
}

.c0:hover {
  background-color: #334bc8;
  color: #ffffff;
}

.c0:focus {
  box-shadow: 0 0 0 3px #8291df;
}

.c0:active {
  background: #8291df;
}

.c0 svg {
  width: 2rem;
  height: 2rem;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c0 svg {
  margin-top: 0.8rem;
}

@media (min-width:1170px) {
  .c0 {
    width: 8.4rem;
    height: 4.2rem;
    border-top-left-radius: 8.5rem;
    border-top-right-radius: 8.5rem;
  }

  .c0 svg {
    width: 2.4rem;
    height: 2.4rem;
  }
}

@media (min-width:1170px) {
  .c0 {
    left: -0.9rem;
  }

  .c0 svg {
    margin-top: 1.5rem;
  }
}

<button
  class="c0"
  direction="left"
>
  <svg
    fill="none"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      aria-label="arrow left"
      d="M8.625 5.25l6.75 6.75-6.75 6.75"
      role="img"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</button>
`;

exports[`<SliderArrow /> should render right arrow 1`] = `
.c0 {
  background-color: #ffffff;
  color: #011fbb;
  width: 6.3rem;
  height: 3rem;
  border-top-left-radius: 6.4rem;
  border-top-right-radius: 6.4rem;
  cursor: pointer;
  border: 0;
  outline: none;
  position: absolute;
  top: 40%;
  bottom: 0;
  -webkit-transition: background-color 0.3s ease-in-out,color 0.1s ease-in-out,visibility 0.3s ease-in-out;
  transition: background-color 0.3s ease-in-out,color 0.1s ease-in-out,visibility 0.3s ease-in-out;
  z-index: 10;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  right: -0.4rem;
}

.c0:hover {
  background-color: #334bc8;
  color: #ffffff;
}

.c0:focus {
  box-shadow: 0 0 0 3px #8291df;
}

.c0:active {
  background: #8291df;
}

.c0 svg {
  width: 2rem;
  height: 2rem;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c0 svg {
  margin-top: 0.8rem;
}

@media (min-width:1170px) {
  .c0 {
    width: 8.4rem;
    height: 4.2rem;
    border-top-left-radius: 8.5rem;
    border-top-right-radius: 8.5rem;
  }

  .c0 svg {
    width: 2.4rem;
    height: 2.4rem;
  }
}

@media (min-width:1170px) {
  .c0 {
    right: -0.9rem;
  }

  .c0 svg {
    margin-top: 1.5rem;
  }
}

<button
  class="c0"
  direction="right"
>
  <svg
    fill="none"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      aria-label="arrow right"
      d="M8.625 5.25l6.75 6.75-6.75 6.75"
      role="img"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</button>
`;
